### PR TITLE
Closes #609

### LINKED ISSUE #609: Create a dataset load for Medical Drugs Sentiment Analysis Corpus
## Adding a Dataset
- **Name:** *Medical Drugs*
- **Description:** *Medical Drugs Sentiment Analysis with Multi-classes*
- **Task:** *Text Classification*
- **Data:** *[data](https://www.kaggle.com/datasets/arbazkhan971/analyticvidhyadatasetsentiment)*
- **License:** *Unknown*
- **Motivation:** *part of InBoXBART training dataset*

### PR DESCRIPTION
**Pull request by shamikbose**
Passes all tests, can be loaded with `load_dataset()`
Some notes:
- Since this task does sentiment classification for a specific medication, this is framed as a Textual Entailment task instead of a Classification
- The test set does not have ground truth, so it's skipped
- Missing `_LICENSE` variable

- **Description:** *This dataset contains comments about patients and the sentiment in those comments about a specific drug that's mentioned*
- **Data:** *https://www.kaggle.com/code/subhankar007/drug-data-sentiment/data*

### Checkbox

- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `biodatasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [ ] Provide values for the ~`_CITATION`~, ~`_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`~, `_LICENSE`, ~`_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`,~ and ~`_BIGBIO_VERSION`~ variables. - *The license is unavailable, so that value is not populated yet*
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `BigBioConfig` for the source schema and one for a bigbio schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_bigbio biodatasets/my_dataset/my_dataset.py`.
- [x] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

**Comment by shamikbose** [comment]
Note: Since this is a Kaggle dataset, it's treated as a local dataset, so it would require a data_dir argument while testing

File: bigbio/biodatasets/medical_data/medical_data.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+        if self.config.data_dir is None:
+            raise ValueError(
+                "This is a local dataset. Please pass the data_dir kwarg to load_dataset."
+            )
+        else:
+            data_dir = self.config.data_dir
+
+        return [
+            datasets.SplitGenerator(
```
Comment (ruisi-su) [inline bigbio/biodatasets/medical_data/medical_data.py:116]: @shamikbose thanks for the quick turnaround on this dataset! When I downloaded this dataset from Kaggle, there was also a test set (`test_tOlRoBf.csv`). Could you please add that split as well?

**Review by ruisi-su — CHANGES_REQUESTED** [review]
@shamikbose I think this dataset would fall into the [text classification](https://github.com/bigscience-workshop/biomedical/blob/master/task_schemas.md#text) schema category (@jason-fries for confirmation). Could you change the schema from TE to TC?

File: bigbio/biodatasets/medical_data/medical_data.py (hunk 1 above)
Comment (shamikbose) [reply bigbio/biodatasets/medical_data/medical_data.py:116]: The test set does not have labels. Here's an example entry

```unique_hash,text,drug
9e9a8166b84114aca147bf409f6f956635034c08,"256 (previously stable on natalizumab), with 55% switching to fingolimod",fingolimod
e747e6822c867571afe7b907b51f0f2ca67b0e1a,On fingolimod and have been since December 2015; the only way I can describe it any better is like feeling drunk but without having had a drink!!!,fingolimod
50b6d851bcff4f35afe354937949e9948975adf7,"Apparently it's shingles! :-/ I do have a few red spots just below my left breast which appeared a few days ago. Wasn't sure if they were connected or not, but showed them to the GP and he immediately said shingles. Apparently 'cos I left it a few days it was too late to prescribe me anything for shingles. The GP prescribed me some useless amitryptiline, which I probably won't take because even a low dose makes me unbearably groggy. Still not entirely happy, just 'cos the stomach pain is so bad and I've had a lot of bloating and stuff. No fever as far as I know. Anyone here had any experience with shingles? Seems like I can expect 3-4 more weeks of this ;_; Dx Crohn's in June 2000. (Yay  ) Tried: 5-ASAs, azathioprine, 6MP, Remicade, methotrexate, Humira, diets. 1st surgery 20/2/13 - subtotal colectomy with end ileostomy. 2nd surgery 10/7/15 - ileorectal anastomosis. Stoma reversed and ileum connected to the rectum. Current status: Chronic flare. Do I have any other kind? Current meds: 50mg 6MP; Entyvio (started 3/11/16)",humira
```

Comment (galtay) [reply bigbio/biodatasets/medical_data/medical_data.py:116]: agree, we should add all splits. it's not uncommon for the test set to be missing some values. @hakunanatasha recently added some flags in the test suite for this https://github.com/bigscience-workshop/biomedical/blob/master/tests/test_bigbio.py#L824

**Comment by shamikbose** [comment]
> @shamikbose I think this dataset would fall into the [text classification](https://github.com/bigscience-workshop/biomedical/blob/master/task_schemas.md#text) schema category (@jason-fries for confirmation). Could you change the schema from TE to TC?
@ruisi-su I left a comment about this in the first commit 
> - Since this task does sentiment classification for a specific medication, this is framed as a Textual Entailment task instead of a Classification

So every text comes with a drug mention and what the text thinks of that specific drug. Putting this in the TE format loses some of that information

File: bigbio/biodatasets/medical_data/medical_data.py (hunk 1 above)
Comment (shamikbose) [reply bigbio/biodatasets/medical_data/medical_data.py:116]: @ruisi-su @galtay Cool, I can use the split parameter. What should the default values be? It's an uint32 field. It can't be 0 since that's a valid value in the train set

Comment (shamikbose) [reply bigbio/biodatasets/medical_data/medical_data.py:116]: @ruisi-su @galtay In the latest commit, changed the sentiment to a `string` type. For missing values in the test split, it's set to `None`

Comment (ruisi-su) [reply bigbio/biodatasets/medical_data/medical_data.py:116]: awesome. That looks good to me. `None` is fine as it is passing the unit tests.

File: bigbio/biodatasets/medical_data/medical_data.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+# limitations under the License.
+
+import os
+from typing import List, Tuple, Dict
+import pandas as pd
+import datasets
+from bigbio.utils import schemas
+from bigbio.utils.configs import BigBioConfig
+from bigbio.utils.constants import Tasks
+
```
Comment (galtay) [inline bigbio/biodatasets/medical_data/medical_data.py:23]: well need a `_LOCAL=True` here .. we haven't added it to the contributing docs yet but all the datasets have it now

Comment (shamikbose) [reply bigbio/biodatasets/medical_data/medical_data.py:23]: Done

File: bigbio/biodatasets/medical_data/medical_data.py (hunk 1 above)
Comment (shamikbose) [reply bigbio/biodatasets/medical_data/medical_data.py:117]: @ruisi-su Does this need any other changes or can it be closed?

File: bigbio/biodatasets/medical_data/medical_data.py (hunk 2 above)
Comment (ruisi-su) [reply bigbio/biodatasets/medical_data/medical_data.py:23]: @galtay is `_LOCAL` the last change this PR needs to be consistent with the latest changes? this PR looks good to me on the split side. It'll be great if you could confirm and then we can merge this one in.
